### ISSUE TITLE
syslog-ng frequent restart on solaris opteron

### ISSUE THREAD

**Issue by lojha** [balabit/syslog-ng#253]
Hello,

We have build syslog-ng 3.4.8 from source on Solaris opteron but when trying to run the syslog-ng daemon it is restarting very frequently. Any help would be highly appreciated.

# ./syslog-ng.d start
syslog-ng service starting.
# ps -ef | grep -i syslog
    root  2458  2456  20 10:55:04 ?           0:05 /opt/soe/local/syslog_ng/sbin/syslog-ng -f /opt/soe/local/syslog_ng/etc/syslog-
    root  2459  2372   1 10:55:08 pts/2       0:00 grep -i syslog
    root  1587     1   0 16:42:55 ?           0:02 /usr/sbin/syslogd
    root  2456     1   1 10:54:51 ?           0:00 /opt/soe/local/syslog_ng/sbin/syslog-ng -f /opt/soe/local/syslog_ng/etc/syslog-
# ps -ef | grep -i syslog
    root  2461  2456  23 10:55:14 ?           0:06 /opt/soe/local/syslog_ng/sbin/syslog-ng -f /opt/soe/local/syslog_ng/etc/syslog-
    root  2462  2372   1 10:55:18 pts/2       0:00 grep -i syslog
    root  1587     1   0 16:42:55 ?           0:02 /usr/sbin/syslogd
    root  2456     1   0 10:54:51 ?           0:00 /opt/soe/local/syslog_ng/sbin/syslog-ng -f /opt/soe/local/syslog_ng/etc/syslog-
# ps -ef | grep -i syslog
    root  2465  2372   1 10:55:29 pts/2       0:00 grep -i syslog
    root  1587     1   0 16:42:55 ?           0:02 /usr/sbin/syslogd
    root  2456     1   0 10:54:51 ?           0:00 /opt/soe/local/syslog_ng/sbin/syslog-ng -f /opt/soe/local/syslog_ng/etc/syslog-
# ps -ef | grep -i syslog
    root  2467  2456  17 10:55:31 ?           0:04 /opt/soe/local/syslog_ng/sbin/syslog-ng -f /opt/soe/local/syslog_ng/etc/syslog-
    root  1587     1   0 16:42:55 ?           0:02 /usr/sbin/syslogd
    root  2456     1   0 10:54:51 ?           0:00 /opt/soe/local/syslog_ng/sbin/syslog-ng -f /opt/soe/local/syslog_ng/etc/syslog-
    root  2468  2372   1 10:55:35 pts/2       0:00 grep -i syslog

**Comment by ihrwein** [comment]
@lojha you can try the `3.6.2` release, too. Unfortunately, without the stack trace we can't identify your problem.

**Comment by ihrwein** [comment]
@lojha Did you manage to try outa newer syslog-ng release? You can also check `3.6.4` as well.

**Comment by ihrwein** [comment]
(The following link can be useful for the other devs: https://www.illumos.org/issues/6474 Maybe it's a Solaris bug.)

**Comment by ihrwein** [comment]
@lojha did you try a newer syslog-ng version?

**Comment by pzoleex** [comment]
Closed due to inactivity

**Comment by mamash** [comment]
This is still an issue for us on SmartOS even with latest syslog-ng-3.13.2 and system ivykis-0.42.2. I'll try to get you a stack trace.

**Comment by mamash** [comment]
The process dumps core every time within the first minute of run time.

```
[root ~]# syslog-ng -e --process-mode=foreground
[2018-01-12T12:54:15.235433] syslog-ng starting up; version='3.13.2'
iv_fd_port_poll: got error 95[Socket operation on non-socket]
Abort (core dumped)

[root ~]# pstack core.syslog-ng.31991
core 'core.syslog-ng.31991' of 31991:	syslog-ng -e --process-mode=foreground
-----------------  lwp# 1 / thread# 1  --------------------
 ffffbf7fff2122fa _portfs () + a
 ffffbf7f4928e909 iv_fd_port_poll () + 149
 ffffbf7f4928ada3 iv_fd_poll_and_run () + 1a3
 ffffbf7f4928c0da iv_main () + 7a
 ffffbf7f48d2568e main_loop_run () + 7e
 00000000004027b1 main () + 1c1
 000000000040229c _start () + 6c
-----------------  lwp# 2 / thread# 2  --------------------
 ffffbf7fff21377d __door_return () + 1d
 ffffbf7fff1f6f0d door_return (0, 0, 0, 0) + cd
 ffffbf7fff2137b0 __door_return () + 50
-----------------  lwp# 3 / thread# 3  --------------------
 ffffbf7fff21377d __door_return () + 1d
 ffffbf7fff1f6f0d door_return (0, 0, 0, 0) + cd
 ffffbf7fff1f75cc door_create_func (0) + 2c
 ffffbf7fff20bb2a _thrp_setup (ffffbf7ffefe9cc0) + 8a
 ffffbf7fff20be40 _lwp_start ()
-----------------  lwp# 4 / thread# 4  --------------------
 ffffbf7fff2130da _lwp_kill () + a
 ffffbf7fff1a64d0 raise (6) + 20
 ffffbf7fff1803a8 abort () + 98
 ffffbf7f49288f50 iv_fatal () + d0
 ffffbf7f4928eba7 iv_fd_port_poll () + 3e7
 ffffbf7f4928ac88 iv_fd_poll_and_run () + 88
 ffffbf7f4928c0da iv_main () + 7a
 ffffbf7f4928a02a iv_work_thread () + 7a
 ffffbf7f4928d364 iv_thread_handler () + 34
 ffffbf7fff20bb2a _thrp_setup (ffffbf7ffefedcc0) + 8a
 ffffbf7fff20be40 _lwp_start ()
```

**Comment by mamash** [comment]
This may or not may not be the same issue though. Looks like running the latest syslog-ng with ivykis-0.40 is fine, but 0.42 is a problem.

**Comment by Kokan** [comment]
@mamash could you do a `git bisect` for ivykis commits between the two version ?

**Comment by mamash** [comment]
Looks like it's related to buytenh/ivykis#12 and specifically to the commit mentioned there that makes linking against pthread optional. When I pick 0.40, or force 0.42.2 to build threadsafe, the issue seems to go away.

**Comment by czanik** [comment]
Is this fix already in pkgsrc for ivykis? Someone is reporting problem on Twitter: https://twitter.com/nuintari/status/962040932026662915

**Comment by mamash** [comment]
It was committed [in January](https://github.com/NetBSD/pkgsrc/commit/6893318e0cd76d970e190251dfecb1a65d940846), but due to the way how the pkgsrc packaging system works, it may take time to make it's way into binary packages.

The last stable quarterly release (2017Q4) was cut before this commit landed. Speaking for SmartOS only, I backported the fix to 2017Q4 too, so the ivykis-0.42.2nb1 in 2017Q4 should be good. Older binary package releases are not.
